### PR TITLE
[PEPC-Boost][1] add types for block assembler rpc + minor refactor

### PR DESCRIPTION
**Pull request by bharath-123**
## 📝 Summary

The block assembler is an important component of PEPC-Boost. The block assembler receives the ToB txs sent by the searchers and the RoB block built by the builders and assembles a final block with the TOB txs on the top and the ROB txs in the remaining blocks. 

This PR defines the request types for the block assembler plus does a minor refactor

### CODE REVIEW

**Review by michaelneuder — APPROVED** [review]
lgtm, unit tests could be nice, but much easier to review

File: eth/block-validation/api_test.go [hunk 1]
```diff
@@ -782,3 +783,90 @@ func TestValidateBuilderSubmissionV2_Blocklist(t *testing.T) {
...
+			Amount:    100,
+			Address:   testAddr,
+		},
+	}
+	execData, err := assembleBlock(api, parent.Hash(), &engine.PayloadAttributes{
+		Timestamp:             parent.Time() + 5,
+		Withdrawals:           withdrawals,
+		SuggestedFeeRecipient: testValidatorAddr,
+	})
+	fmt.Printf("DEBUG: Exec data random is %x\n", execData.Random)
```
Comment (michaelneuder) [inline eth/block-validation/api_test.go:832]: remove ?

**Review by michaelneuder — APPROVED** [review]
nice!

File: eth/block-validation/api_test.go (hunk 1 above)
Comment (bharath-123) [reply eth/block-validation/api_test.go:832]: yeah dude will remove these in all PRs

**Pull request merged**